docs/documentation.md

```diff
@@ -0,0 +1,424 @@
+```
+cleanupExpiredListings Cleanup the expired listing by iterating over the provided range of indexes.
+
+---
+
+### fun `cleanupPurchasedListings()`
+
+```cadence
+fun cleanupPurchasedListings(listingResourceID: UInt64)
+```
+cleanupPurchasedListings
+Allows anyone to remove already purchased listings.
+
+---
+
+### fun `getExistingListingIDs()`
+
+```cadence
+fun getExistingListingIDs(nftType Type, nftID UInt64): [UInt64]
+```
+getExistingListingIDs
+Returns an array of listing IDs of the given `nftType` and `nftID`.
+
+---
+
+## Events
+
+### event `StorefrontInitialized`
+
+```cadence
+event StorefrontInitialized(storefrontResourceID: UInt64)
+```
+A Storefront resource has been created. Consumers can now expect events from this Storefront. Note that we do not specify an address: we cannot and should not. Created resources do not have an owner address, and may be moved
+after creation in ways we cannot check. `ListingAvailable` events can be used to determine the address
+of the owner of the Storefront at the time of the listing but only at that precise moment in that precise transaction. If the seller moves the Storefront while the listing is valid, that is on them.
+
+---
+
+### event `StorefrontDestroyed`
+
+```cadence
+event StorefrontDestroyed(storefrontResourceID: UInt64)
+```
+A Storefront has been destroyed. Event consumers can now stop processing events from this Storefront.
+Note - we do not specify an address.
+
+---
+
+### event `ListingAvailable`
+
+```cadence
+event ListingAvailable(storefrontAddress: Address, listingResourceID: UInt64, nftType: Type, nftUUID: UInt64, nftID: UInt64, salePaymentVaultType: Type, salePrice: UFix64, customID: String?, commissionAmount: UFix64, commissionReceivers: [Address]?, expiry: UInt64)
+```
+
+Above event gets emitted when a listing has been created and added to a Storefront resource. The Address values here are valid when the event is emitted, but the state of the accounts they refer to may change outside of the
+NFTStorefrontV2 workflow, so be careful to check when using them.
+
+---
+
+### event `ListingCompleted`
+
+```cadence
+event ListingCompleted(listingResourceID: UInt64, storefrontResourceID: UInt64, purchased: Bool, nftType: Type, nftUUID: UInt64, nftID: UInt64, salePaymentVaultType: Type, salePrice: UFix64, customID: String?, commissionAmount: UFix64, commissionReceiver: Address?, expiry: UInt64)
+```
+The listing has been resolved. It has either been purchased, removed or destroyed.
+
+---
+
+### event `UnpaidReceiver`
+
+```cadence
+event UnpaidReceiver(receiver: Address, entitledSaleCut: UFix64)
+```
+A entitled receiver has not been paid during the sale of the NFT.
+
+---
+
+
+**Holistic process flow diagram of NFTStorefrontV2 -** 
+
+[NFT Storefront Process flow | Lucidchart](https://lucid.app/lucidchart/a0015c93-183d-4ab5-8a28-8909cc34b25d/edit?viewport_loc=-565%2C-377%2C3743%2C1941%2Cm-5o7ONTd-nK&invitationId=inv_f6e78c34-46da-41b4-a229-95eae82e8e9f#)
```

Comment:
We should probably only post the JPG of the LucidChart rather than the LucidChart itself :)